modules/bitgo/src/v2/coins/near.ts

```diff
@@ -269,13 +340,274 @@ export class Near extends BaseCoin {
+    }
+    const userSigningMaterial = JSON.parse(userPrv) as UserSigningMaterial;
+
+    let backupPrv;
+    try {
+      backupPrv = this.bitgo.decrypt({
+        input: backupKey,
+        password: params.walletPassphrase,
+      });
+    } catch (e) {
+      throw new Error(`Error decrypting backup keychain: ${e.message}`);
+    }
+    const backupSigningMaterial = JSON.parse(backupPrv) as BackupSigningMaterial;
+    /* ********************** END ***********************************/
+
+    await Near.initializeMPCIfNotPresent();
+    const accountId = Near.MPC.deriveUnhardened(bitgoKey, `m/0`).slice(0, 64);
+    const bs58EncodedPublicKey = nearAPI.utils.serialize.base_encode(new Uint8Array(Buffer.from(accountId, 'hex')));
+    const availableBalance = new BigNumber(await this.getAccountBalance(accountId));
+    const { nonce, blockHash } = await this.getAccessKey({ accountId, bs58EncodedPublicKey });
+    const gasPrice = await this.getGasPrice(blockHash);
+    const gasPriceFirstBlock = new BigNumber(gasPrice);
+    const gasPriceSecondBlock = gasPriceFirstBlock.multipliedBy(1.05);
+    const { transfer_cost, action_receipt_creation_config } = NearGasConfigs;
+    const totalGasRequired = transfer_cost.send_sir
+      .plus(action_receipt_creation_config.send_sir)
+      .multipliedBy(gasPriceFirstBlock)
+      .plus(transfer_cost.execution.plus(action_receipt_creation_config.execution).multipliedBy(gasPriceSecondBlock));
+    // adding some padding to make sure the gas doesn't go below required gas by network
+    const totalGasWithPadding = totalGasRequired.multipliedBy(1.5);
+    const netAmount = availableBalance.minus(totalGasWithPadding).toFixed();
+    const factory = accountLib.register(this.getChain(), accountLib.Near.TransactionBuilderFactory);
+    const txBuilder = factory
+      .getTransferBuilder()
+      .sender(accountId, accountId)
+      .nonce(nonce)
+      .receiverId(params.recoveryDestination)
+      .recentBlockHash(blockHash)
+      .amount(netAmount);
+    const unsignedTransaction = (await txBuilder.build()) as accountLib.Near.Transaction;
+
+    // add signature
+    const signatureHex = await this.getTSSSignature(
+      userSigningMaterial,
+      backupSigningMaterial,
+      'm/0',
+      unsignedTransaction
+    );
+    const publicKeyObj = { pub: accountId };
+    txBuilder.addSignature(publicKeyObj as PublicKey, signatureHex);
+    const signedTransaction = await txBuilder.build();
+    const serializedTx = signedTransaction.toBroadcastFormat();
+    return { serializedTx: serializedTx };
+  }
+
+  /**
+   * Make a request to one of the public EOS nodes available
+   * @param params.endpoint
+   * @param params.payload
+   */
+  protected async getDataFromNode(params: { payload?: Record<string, unknown> }): Promise<request.Response> {
+    const nodeUrls = this.getPublicNodeUrls();
+    for (const nodeUrl of nodeUrls) {
+      try {
+        return await request.post(nodeUrl).send(params.payload);
```

Comment:
Instead of serially trying out each nodeurl at a time, may be try using promise.any, refer [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/any) , might be the right alternative if I understood the logic correctly

Reply:
As discussed.